### PR TITLE
Story navigation education.

### PR DESCRIPTION
**Pull request by gmajoulet**
Work on the `amp-story-education` extension:

- Lazy loading the `amp-story-education` extension when the `#cap=education` parameter is present
- Tap and swipe education screens
- Navigation between screens, and hide education when done
- Pausing/unpausing the story
- RTL support
- Shadow DOM
- i18n
- unit tests

Not included: animations, viewer messaging.

#27097

### CODE REVIEW

**Comment by gmajoulet** [comment]
Friendly ping, it'd be great to have this one (and the two followups) in tomorrow's canary cut.

cc @dvoytenko for `src/localized-strings.js` and `build-system/test-configs/dep-check-config.js`

**Comment by newmuis** [comment]
Weird edge case: have you tested this on a story with consent enabled?

**Comment by gmajoulet** [comment]
Yes, the `z-index` and logic around the `pausedState` handle it nicely :))

**Comment by gmajoulet** [comment]
(To clarify: the education comes before the consent)

File: extensions/amp-story/1.0/amp-story.js [hunk 1]
```diff
@@ -2264,6 +2266,21 @@ export class AmpStory extends AMP.BaseElement {
       });
   }
 
+  /**
+   * Loads amp-story-education if the viewer capability is provided.
+   * @private
+   */
+  maybeLoadStoryEducation_() {
+    if (!this.viewer_.hasCapability('education')) {
```
Comment (newmuis) [inline extensions/amp-story/1.0/amp-story.js]: Sorry, I think I'm not actually sure what this buys us after all.  There might be some experiences that we think old viewers should get.

If the viewer doesn't support education, it won't respond to the education viewer messages.  So I feel like we'd need this only if we want to differentiate between viewers that support education but don't respond vs viewers that don't support education.  But we don't need to distinguish that, right?

**Comment by newmuis** [comment]
Ugh, GH lost my other comment: is this just for the onboarding screens?  i.e. if we were to implement another set of screens that don't show up on the initial load, we would need another separate extension?

File: extensions/amp-story/1.0/amp-story.js (hunk 1 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story.js]: It mostly allows lazy loading actually, so we don't have to load the extension all the time.

> is this just for the onboarding screens? i.e. if we were to implement another set of screens that don't show up on the initial load, we would need another separate extension?

This PR doesn't show any screen at all actually. It loads the extension, that will then, in a followup PR, start sending viewer messages to know if it should display an education screen.
We'll be in control of when to send these messages, on initial load or anything else (the extension has access to the store service).

File: extensions/amp-story/1.0/amp-story.js [hunk 2]
```diff
@@ -2264,6 +2266,21 @@ export class AmpStory extends AMP.BaseElement {
...
+      return;
+    }
+
+    Services.extensionsFor(this.getAmpDoc().win).installExtensionForDoc(
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story.js]: I'm not entirely sure this actually works for all cases, I'm currently investigating a bug where the `viewer-integration.js` extension in the open source player isn't loaded in canonical sites. I think it might have to do with the `this.getAmpDoc().win` bit. I think we should replace it with this `this.win`, but I'm still in the process of verifying.

Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story.js]: Cool, thanks. It looks like all other extensions use `this.win`, I replaced both.
